compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConstDeclarationChecker.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.lexer.KtTokens
+
+object FirConstDeclarationChecker : FirBasicDeclarationChecker() {
+    override fun check(declaration: FirDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (declaration !is FirVariable<*>) return
+        val source = declaration.source ?: return
+        if (source.kind is FirFakeSourceElementKind) return
+        val modifierList = with(FirModifierList) { source.getModifierList() }
+        val constModifier = modifierList?.modifiers?.singleOrNull { it.token == KtTokens.CONST_KEYWORD } ?: return
+
+        if (declaration.isVar) {
+            reporter.reportOn(constModifier.source, FirErrors.WRONG_MODIFIER_TARGET, constModifier.token, "vars", context)
+        }
+
+        // TODO: Implement checkers for these errors (see ConstModifierChecker in FE1.0):
+        // - CONST_VAL_NOT_TOP_LEVEL_OR_OBJECT
```

Comment:
Mark, I should mention that all these CONST_* checkers are already in progress.

Reply:
Oh I see them in the diagnostic list, sorry for not checking before. I guess the amount of conflict depends where/how it will be implemented. Do you want me to just remove the comments then?

Reply:
See https://jetbrains.quip.com/bON4AjCwIJaE/FIR-diagnostic-list (Taken-by-interns list)

Reply:
> Do you want me to just remove the comments then?

No, comments aren't a problem. Depending on how this new checker will be implemented, I will either merge it into your class, or just remove the comments from it.

Reply:
> I guess the amount of conflict depends where/how it will be implemented.

I expect to have them at the end of this week or may be next week.

Reply:
Today I noticed that your checker works with `FirDeclaration` (and it's correct because of diagnostic kind). However, as I can see now all other checkers looks better extending `FirPropertyChecker`, so, probably, it's better to declare something like `FirConstPropertyChecker` for them. I will re-check this again, and add your TODO comments in this `FirConstPropertyChecker` instead.

Reply:
I have just added https://github.com/JetBrains/kotlin/blob/master/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConstPropertyChecker.kt from another PR and your comments about TODO are included there, so please remove comments from your class // cc @punzki

Reply:
Looks like I can simply move it to `FirConstPropertyChecker` so I'm deleting this file.